Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -122,7 +122,13 @@ void UPSWorldSubsystem::RegisterSpotComponent(UPSSpotComponent* MySpotComponent)
 void UPSWorldSubsystem::OnInitialized_Implementation()
 {
 	StarDynamicProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
-	if (!ensureMsgf(StarDynamicProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicProgressMaterial' is null!"), __LINE__, __FUNCTION__))
+	StarLockedProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
+	StarUnLockedProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
+
+
+	if (!ensureMsgf(StarDynamicProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicProgressMaterial' is null!"), __LINE__, __FUNCTION__)
+		|| !ensureMsgf(StarLockedProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarLockedProgressMaterial' is null!"), __LINE__, __FUNCTION__)
+		|| !ensureMsgf(StarUnLockedProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarUnLockedProgressMaterial' is null!"), __LINE__, __FUNCTION__))
```

Comment:
Please remove these 3 `ensure` statements that verify material instances - their references are not used in this body, so no need for ensure. You already have good `ensure` inside `SetStarActorProgressMeshMaterial(Material)`, where it's actually used, and another one I asked to add for the material itself in [this comment](https://github.com/moinm3uw/ProgressionSystem/pull/55/files#r1953744065).

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/55/commits/34dd8540afb623b4aa721b4cdfe6277a0b8a5646